pkg/provisioner/ironic/ironic.go

```diff
@@ -629,7 +629,7 @@ func (p *ironicProvisioner) changeNodeProvisionState(ironicNode *nodes.Node, opt
 // details of devices discovered on the hardware. It may be called
 // multiple times, and should return true for its dirty flag until the
 // inspection is completed.
-func (p *ironicProvisioner) InspectHardware(force bool) (result provisioner.Result, details *metal3v1alpha1.HardwareDetails, err error) {
+func (p *ironicProvisioner) InspectHardware(force, forceNew bool) (result provisioner.Result, details *metal3v1alpha1.HardwareDetails, err error) {
```

Comment:
I wanted to use `force` parameter, but I thought I would break the existing functionally. But if you think it is okay to use it, then I will change the code to rely on it, instead of adding `forceNew`

Reply:
I think it's a possibility worth to be explored to avoid increasing the complexity of the function signature (and moreover the semantic). Enriching unit tests on this portion of code could help to ensure not to break the existing functionality

Reply:
Removed `forceNew`.